File: src/Bionic.Bridge.Capacitor/HapticsBridge.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Bionic.Bridge.Capacitor {
    public static class HapticsBridge {
        private const string HapticsVibrateFunctionName = "BionicBridge.Capacitor.Haptics.vibrate";
        private const string HapticsImpactFunctionName = "BionicBridge.Capacitor.Haptics.impact";
        private const string HapticsSelectionStartFunctionName = "BionicBridge.Capacitor.Haptics.selectionStart";
        private const string HapticsSelectionChangedFunctionName = "BionicBridge.Capacitor.Haptics.selectionChanged";
        private const string HapticsSelectionEndFunctionName = "BionicBridge.Capacitor.Haptics.selectionEnd";

        public static Task Vibrate() =>
            JSRuntime.Current.InvokeAsync<object>(HapticsVibrateFunctionName);

        public static Task Impact(HapticsImpactStyle style) =>
            JSRuntime.Current.InvokeAsync<object>(HapticsImpactFunctionName, style);

        public static Task SelectionStart() =>
            JSRuntime.Current.InvokeAsync<object>(HapticsSelectionStartFunctionName);

        public static Task SelectionChanged() =>
            JSRuntime.Current.InvokeAsync<object>(HapticsSelectionChangedFunctionName);

        public static Task SelectionEnd() =>
            JSRuntime.Current.InvokeAsync<object>(HapticsSelectionEndFunctionName);
    }

    public enum HapticsImpactStyle {
        Heavy,
        Medium,
        Light
    }
}

File: src/Bionic.Bridge.Capacitor.JS/src/HapticsBridge.ts
import {HapticsImpactStyle, Plugins} from "@capacitor/core";

const {Haptics} = Plugins;

interface IHaptics {
  vibrate(): Promise<void>;
  impact(style: number): Promise<void>;
  selectionStart(): Promise<void>;
  selectionChanged(): Promise<void>;
  selectionEnd(): Promise<void>;
}

export class HapticsBridge implements IHaptics {
  async vibrate(): Promise<void> {
    if (!Haptics) return Promise.reject("Haptics not available.");
    await Haptics.vibrate();
  }

  async impact(styleOption: number): Promise<void> {
    if (!Haptics) return Promise.reject("Haptics not available.");
    let style = HapticsImpactStyle.Light;
    switch (styleOption) {
      case 0:
        style = HapticsImpactStyle.Heavy;
        break;
      case 1:
        style = HapticsImpactStyle.Medium;
        break;
    }
    await Haptics.impact({style});
  }

  async selectionStart(): Promise<void> {
    if (!Haptics) return Promise.reject("Haptics not available.");
    await Haptics.selectionStart();
  }

  async selectionChanged(): Promise<void> {
    if (!Haptics) return Promise.reject("Haptics not available.");
    await Haptics.selectionChanged();
  }

  async selectionEnd(): Promise<void> {
    if (!Haptics) return Promise.reject("Haptics not available.");
    await Haptics.selectionEnd();
  }
}
